src/ShelfLayoutManager.Web/Controllers/CabinetsController.cs

```diff
@@ -0,0 +1,44 @@
+using FluentResults;
+using Microsoft.AspNetCore.Mvc;
+using ShelfLayoutManager.Core;
+using ShelfLayoutManager.Infrastructure;
+
+namespace ShelfLayoutManager.Web.Controllers;
+
+[ApiController]
+[Route("v1/cabinets")]
+public class CabinetsController(ICabinetsDataService cabinetsDataService)
+    : ControllerBase
+{
+    private readonly ICabinetsDataService _cabinetsDataService = cabinetsDataService;
+
+    [HttpGet("{id}")]
+    public async Task<ActionResult<Cabinet?>> GetAsync(
+        [FromRoute] long id,
+        [FromQuery] bool includeRows = false,
+        [FromQuery] bool includeLanes = false,
+        CancellationToken ct = default)
+    {
+        Result<Cabinet?> cabinetResult = await _cabinetsDataService.GetAsync(id, includeRows, includeLanes, ct);
+
+        if (cabinetResult.Value is null)
+        {
+            return NotFound();
+        }
+
+        return cabinetResult.Value;
+    }
+
+    [HttpGet]
+    public async Task<IList<Cabinet>> ListAsync(
+        [FromQuery] bool includeRows = false, [FromQuery] bool includeLanes = false, CancellationToken ct = default)
+            => (await _cabinetsDataService.ListAsync(includeRows, includeLanes, ct)).Value;
+
+    [HttpPost("-/assign-sku")]

```

Comment:
Nit: A bit weird to have singular dash, `.../-/...` as part of a path

Reply:
This is also probably the most overloaded endpoint in the codebase.
It has 3 responsibilities.
1. Set SKU assignment
2. Remove SKU assignment
3. Move SKU assignment

These could really be 3 separate endpoints and be less confusing to look at.
And easier to validate.

Reply:
Wasn't sure where abstraction was broken up.
That abstraction went all the way down to service class.
Used if statements in service class to check which operation to run